Add > Steampipe Cloud > Integrations > Doc > Golang (#58)
turbot/steampipe-docs: docs/cloud/integrations/golang.md

---
title:  Connect to Steampipe Cloud with Golang
sidebar_label: Golang
---
# Connect to Steampipe Cloud from Golang

> Since your Steampipe Cloud workspace is a Postgres database, by using the `pq` driver you can connect and query your workspace database with Golang.

Reviewer: ```suggestion
Since your Steampipe Cloud workspace is a Postgres database, you can connect and query using Go's `pq` driver.
```

Author: Rectified

> The [Connect](/docs/cloud/integrations/overview) tab for your workspace provides the details you need to connect Steampipe Cloud with Golang.

Reviewer: ```suggestion
The [Connect](/docs/cloud/integrations/overview) tab for your workspace provides the details you need to connect from Go.
```

Author: Rectified

<div style={{"marginTop":"1em", "marginBottom":"1em", "width":"90%"}}>
<img src="/images/docs/cloud/steampipe-cloud-connect-details.jpg" />
</div>

> You'll need the Postgres driver for Go's database/sql package, which you can install using `go get github.com/lib/pq`.

Reviewer: ```suggestion
You'll need the Postgres driver for Go's `database/sql` package, which you can install using `go get github.com/lib/pq`.
```

Author: Rectified

Then you specify the connection string, create a connection, run a query, and fetch results. In this example, we query the name, region and the versioning state of `aws_s3_bucket`.

```go
package main

import (
	"database/sql"
	"fmt"
	_ "github.com/lib/pq"
)

var db *sql.DB
func main() {
var err error

connStr := "postgresql://rahulsrivastav14:f3**-****-**2c@rahulsrivastav14-rahulsworkspace.usea1.db.steampipe.io:9193/dea4px"
db, err = sql.Open("postgres", connStr)
if err != nil {
panic(err)
}
if err = db.Ping(); err != nil {
panic(err)
}
rows, err := db.Query(`select name, region, versioning_enabled from aws_s3_bucket`)
defer rows.Close()

var name string
var region string
var versioning_enabled string

	for rows.Next() {
	switch err := rows.Scan(&name, &region, &versioning_enabled); err {
	case sql.ErrNoRows:
		fmt.Println("No matching rows")
	case nil:
		fmt.Println(name, region, versioning_enabled, "\n")
	default:
		panic(err)
		}
	}

> }

Reviewer: The formatting is off here. Go has a lovely approach: You run `gofmt -s -w .` and it decides how things should be.

Author: Thank you for sharing the command. I've formatted the code and pasted it to the doc.

```

```
aws-glue-temporary-986325076436-us-east-1 us-east-1 false

aws-logs-986325076436-us-east-1 us-east-1 true

turbot-986325076436-us-east-1 us-east-1 true

appstream2-36fb080bb8-us-east-1-986325076436 us-east-1 true

integratedtagsbucket2022 us-east-1 false

```

## Connect to Steampipe CLI from Golang

> To connect Golang to [Steampipe CLI](https://steampipe.io/downloads), run `steampipe service start --show-password` and use the displayed connection details.

Reviewer: ```suggestion
To connect to [Steampipe CLI](https://steampipe.io/downloads), run `steampipe service start --show-password` and use the displayed connection details.
```

Author: Rectified

```
Steampipe service is running:
Database:
  Host(s):            localhost, 127.0.0.1, 192.168.29.204
  Port:               9193
  Database:           steampipe
  User:               steampipe
  Password:           99**_****_**8c
  Connection string:  postgres://steampipe:99**_****_**8c@localhost:9193/steampipe
```

## Call the Steampipe Cloud API from Golang

> You can also use the [Steampipe Cloud query API](https://steampipe.io/docs/cloud/develop/query-api) with the Golang `net/http` driver. Grab your [token](https://steampipe.io/docs/cloud/profile#api-tokens), put it an environment variable like `STEAMPIPE_CLOUD_TOKEN`, and call this variable using Go's `os` driver.

Reviewer: ```suggestion
You can also use the [Steampipe Cloud query API](https://steampipe.io/docs/cloud/develop/query-api) with Go's `net/http` driver. Grab your [token](https://steampipe.io/docs/cloud/profile#api-tokens), put it an environment variable like `STEAMPIPE_CLOUD_TOKEN`, and make an HTTP request.
```

Author: Rectified.
